src/mad_spark_alt/evolution/semantic_operators.py

```
@@ -0,0 +1,532 @@
+        mutated_content: str,
+        llm_cost: float,
+        mutation_type: Optional[str] = None
+    ) -> GeneratedIdea:
+        """Create a mutated idea object."""
+        return GeneratedIdea(
+            content=mutated_content,
+            thinking_method=original.thinking_method,
+            agent_name="BatchSemanticMutationOperator",
+            generation_prompt=f"Semantic mutation of: '{original.content[:50]}...'",
+            confidence_score=(original.confidence_score or 0.5) * 0.95,
+            reasoning=f"Applied semantic mutation to create variation",
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)